src/pages/Facility/settings/organizations/FacilityOrganizationIndex.tsx

```diff
@@ -43,16 +46,152 @@ export default function FacilityOrganizationIndex({
   facilityId: string;
 }) {
   const [expandedRows, setExpandedRows] = useState<Record<string, boolean>>({});
+  const [searchQuery, setSearchQuery] = useState("");
   const { t } = useTranslation();
+  const isMobile = useBreakpoints({ default: true, sm: false });
+
   const { data, isLoading } = useQuery({
     queryKey: ["facilityOrganization", "list", facilityId],
     queryFn: query.paginated(routes.facilityOrganization.list, {
       pathParams: { facilityId },
     }),
     enabled: !!facilityId,
   });
+
   const tableData = data?.results || [];
-  const isMobile = useBreakpoints({ default: true, sm: false });
+
+  const [{ childrenMap, topLevelOrganizations }, setOrganizationHierarchy] =
+    useState<{
+      childrenMap: Map<string, (typeof tableData)[0][]>;
+      topLevelOrganizations: (typeof tableData)[0][];
+    }>({ childrenMap: new Map(), topLevelOrganizations: [] });
+
+  useEffect(() => {
+    const childrenMap = new Map<string, (typeof tableData)[0][]>();
+    const topLevelOrgs = tableData.filter(
+      (org) => !org.parent || Object.keys(org.parent).length === 0,
+    );
+
+    tableData.forEach((org) => {
+      if (org.parent?.id) {
+        const parentId = org.parent.id;
+        if (!childrenMap.has(parentId)) {
+          childrenMap.set(parentId, []);
+        }
+        childrenMap.get(parentId)?.push(org);
+      }
+    });
+
+    setOrganizationHierarchy({
+      childrenMap,
+      topLevelOrganizations: topLevelOrgs,
+    });
+  }, [tableData]);
+
+  const getChildren = useCallback(
+    (parentId: string) => {
+      return childrenMap.get(parentId) || [];
+    },
+    [childrenMap],
+  );
+
+  const createSearchMatcher = (query: string) => {
+    const normalizedQuery = query.toLowerCase().trim();
+    return (name: string) => name.toLowerCase().includes(normalizedQuery);
+  };
+
+  const matchesSearch = useMemo(
+    () => createSearchMatcher(searchQuery),
+    [searchQuery],
+  );
+
+  const hasMatchingChildren = useCallback(
+    (parentId: string): boolean => {
+      const children = childrenMap.get(parentId) || [];
+      return children.some(
+        (child) => matchesSearch(child.name) || hasMatchingChildren(child.id),
+      );
+    },
+    [childrenMap, matchesSearch],
+  );
+
+  const filteredData = useMemo(() => {
```

Comment:
I think I missed this when reviewing Locations 😅 

Both useMemo and useCallback are unnecessary - the data here would never be large enough to warrant their usage; performance wise if anything maybe worse due to overhead of using these.

We can add it back later if there's noticeable performance issues.

Feel free to make the change in Locations as well, but it's not a priority.